public/app/features/scenes/querying/SceneQueryRunner.ts

```diff
@@ -174,6 +209,53 @@ export class SceneQueryRunner extends SceneObjectBase<QueryRunnerState> {
   private onDataReceived = (data: PanelData) => {
     this.setState({ data });
   };
+
+  private applyPanelTimeOverrides(panel: { timeFrom?: string; timeShift?: string }, timeRange: TimeRange): TimeRange {
+    let newTimeRange = timeRange;
+
+    if (panel.timeFrom) {
+      const timeFromInterpolated = sceneGraph.interpolate(this, panel.timeFrom);
+      const timeFromInfo = rangeUtil.describeTextRange(timeFromInterpolated);
+      if (timeFromInfo.invalid) {
+        return newTimeRange;
+      }
+
+      if (typeof timeRange.raw.from === 'string') {
+        const timeFromDate = dateMath.parse(timeFromInfo.from)!;
+        newTimeRange = {
+          from: timeFromDate,
+          to: dateMath.parse(timeFromInfo.to)!,
+          raw: {
+            from: timeFromInfo.from,
+            to: timeFromInfo.to,
+          },
+        };
+      }
+    }
+
+    if (panel.timeShift) {
+      const timeShiftInterpolated = sceneGraph.interpolate(this, panel.timeShift);
+      const timeShiftInfo = rangeUtil.describeTextRange(timeShiftInterpolated);
+      if (timeShiftInfo.invalid) {
+        return newTimeRange;
+      }
+
+      const timeShift = '-' + timeShiftInterpolated;
+      const from = dateMath.parseDateMath(timeShift, newTimeRange.from, false)!;
+      const to = dateMath.parseDateMath(timeShift, newTimeRange.to, true)!;
+
+      newTimeRange = {
+        from,
+        to,
+        raw: {
+          from,
+          to,
+        },
+      };
+    }
+
+    return newTimeRange;
+  }
```

Comment:
I think you can just reuse the existing function applyPanelTimeOverrides with some small refactoring (pass in interpolate function) 
https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/utils/panel.ts#L114 

by splitting  getTimeOverrideInfo and  applyPanelTimeOverrides it looks like you are duplicating a lot logic.

Reply:
But on second thought, as we are going to move this to the scenes package, we would need to move applyPanelTimeOverrides as well, so it's ok if it's duplicated, hopefully, we can get rid of the old next year.

But I think doing both the new time range and the override info in the same function is cleaner / less duplicated logic like https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/utils/panel.ts#L114 is still better?

Reply:
Yep, they were split because they were living in different places. After your suggestion, both are in SceneQueryRunner, so it make sense to have them together 👍

Reply:
And yes, I duplicated it on purpose as well as we're duplicating the variable "get option" logic. To not create dependencies with core. So yep, I agree 😄